octopoes/octopoes/api/api.py

```diff
@@ -57,47 +58,56 @@
 
 
 @app.exception_handler(RequestValidationError)
-def http_validation_exception_handler(request: Request, exc: RequestValidationError) -> JSONResponse:
-    logger.critical(exc)
-    return JSONResponse(
-        {
-            "value": str(exc),
-        },
-        status.HTTP_422_UNPROCESSABLE_ENTITY,
-    )
+def http_validation_exception_handler(_: Request, exc: RequestValidationError) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(exc))
 
 
 @app.exception_handler(RequestError)
-def http_exception_handler(request: Request, exc: RequestError) -> JSONResponse:
-    logger.critical(exc)
-    return JSONResponse(
-        {
-            "value": str(exc),
-        },
-        status.HTTP_502_BAD_GATEWAY,
-    )
+def http_exception_handler(_: Request, exc: RequestError) -> None:
+    logger.error(exc)
+    raise HTTPException(status.HTTP_502_BAD_GATEWAY, detail=str(exc))
+
+
+@app.exception_handler(InvalidField)
+def invalid_field(_: Request, exc: InvalidField) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(InvalidPath)
+def invalid_path(_: Request, exc: InvalidPath) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(ValueError)
+def value_error(_: Request, exc: ValueError) -> None:
+    logger.info(exc)
+    raise HTTPException(status.HTTP_400_BAD_REQUEST, detail=str(exc))
+
+
+@app.exception_handler(TypeNotFound)
+def type_not_found(_: Request, exc: TypeNotFound) -> None:
+    raise HTTPException(status.HTTP_404_NOT_FOUND, detail="Type not found")
```

Comment:
We don't have an endpoint that requests a type for which a 404 would make sense if the type doesn't exist. The type is part of a query for example and I think it would make more sense to return a 400 bad request if a type that doesn't exist is specified.